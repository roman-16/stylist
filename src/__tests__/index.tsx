import { render } from '@testing-library/react';
import { createStyled } from '@';

const styled = createStyled();
const Input = styled('input', {
  display: 'flex',
  fontSize: '16px',
  '&:hover': {
    fontSize: '8px',
  },
});

it('creates a Input component', () => {
  const { container } = render(<Input />);
  const input = container.children[0] as HTMLInputElement;

  expect(container).toMatchSnapshot();
  expect(input).toHaveStyle({
    display: 'flex',
    fontSize: '16px',
  });
  expect(Array.from(document.styleSheets[0].cssRules).map(({ cssText }) => cssText)).toMatchSnapshot();
});

it('creates a component with custom csss', () => {
  const { container: inputContainer } = render(<Input />);
  const { container } = render(<Input css={{ margin: '16px' }} />);

  expect(inputContainer.children[0]).toHaveStyle({
    display: 'flex',
    fontSize: '16px',
    margin: undefined,
  });
  expect(container.children[0]).toHaveStyle({
    display: 'flex',
    fontSize: '16px',
    margin: '16px',
  });
});

it('lets you add attributes', () => {
  const InputWithId = Input.attrs({
    id: 'test-id',
  });
  const { container } = render(<InputWithId />);

  expect(container).toMatchSnapshot();
});

it('lets you use an empty styles object', () => {
  const Box = styled('div', {});
  const { container } = render(<Box />);

  expect(container).toMatchSnapshot();
});

it('creates an extended component', () => {
  const InputMargin = styled(Input, {
    fontSize: undefined,
    margin: '16px',
  });

  const { container: inputContainer } = render(<Input />);
  const { container } = render(<InputMargin />);

  expect(inputContainer.children[0]).toHaveStyle({
    display: 'flex',
    fontSize: '16px',
    margin: undefined,
  });
  expect(container.children[0]).toHaveStyle({
    display: 'flex',
    fontSize: undefined,
    margin: '16px',
  });
});
