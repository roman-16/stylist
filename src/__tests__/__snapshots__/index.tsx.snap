// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a Input component 1`] = `
<div>
  <input
    class="sts-14765"
  />
</div>
`;

exports[`creates a Input component 2`] = `
Array [
  ".sts-14765 {display: flex; font-size: 16px;}",
  ".sts-14765:hover {font-size: 8px;}",
]
`;

exports[`lets you add attributes 1`] = `
<div>
  <input
    class="sts-14765"
    id="test-id"
  />
</div>
`;

exports[`lets you use an empty styles object 1`] = `
<div>
  <div
    class="sts-34788"
  />
</div>
`;
